internal/engines/check.go

```
@@ -166,53 +175,71 @@ func (engine *CheckEngine) check(
-	// and return the permission if the first CheckFunction succeeds and all others fail (exclusion).
-	case *base.Rewrite_OPERATION_EXCLUSION.Enum():
-		return engine.setChild(ctx, request, rewrite.GetChildren(), checkExclusion)
-	// In case of an undefined child type, return a CheckFunction that always fails.
-	default:
-		return checkFail(errors.New(base.ErrorCode_ERROR_CODE_UNDEFINED_CHILD_TYPE.String()))
+	path := coverage.PathFromContext(ctx)
+	return func(ctx context.Context) (*base.PermissionCheckResponse, error) {
+		trackCtx := coverage.ContextWithRegistry(ctx, engine.registry)
+		trackPath := coverage.GetPath(request.GetPermission(), request.GetMetadata().GetExclusionPath())
+		coverage.Track(coverage.ContextWithPath(trackCtx, trackPath))
+		
```

golangci-lint (2.5.0)

[error] File is not properly formatted

(gofumpt)